weasyl/searchtag.py

```diff
@@ -217,3 +265,186 @@ def tag_history(submitid):
         .select_from(tu.join(pr, tu.c.userid == pr.c.userid))
         .where(tu.c.submitid == submitid)
         .order_by(tu.c.updated_at.desc()))
+
+
+def edit_searchtag_blacklist(userid, tags, edit_global_blacklist=False):
```

Comment:
Time for the big annoying change: split this into separate functions for editing user and global blacklists, and have neither of them deal with determining which tags have to be newly created, putting *that* in yet another function. And maybe reusing it in `associate`, too. Fun, right?

I can do this if you don’t want to. :D